corehq/apps/user_importer/importer.py

```diff
@@ -578,3 +580,127 @@ def _get_domain_info(domain):
             ret['errors'].append(_error_message)
 
     return ret
+
+
+def create_or_update_web_users(upload_domain, user_specs, upload_user, update_progress=None):
```

Comment:
Is this copied from the combined web and mobile worker one above? If so it would be nice to break the shared functionality into a function, and call it from both places, rather than duplicate the code.

Reply:
Similar to the views, it looks like much of this functionality is shared with both functions (the handling of web users, handling locations, pulling info from the rows), I would guess this can be combined into one function, with an argument passed to show if its web or mobile, but if not, those parts could at least be pulled into helper functions and called in both places.